### PR TITLE
Cancel button will revert displayed color back to previous color

### PR DESCRIPTION
**Pull request by Blackbaud-AlexKingman**

### CODE REVIEW

File: src/app/public/modules/colorpicker/colorpicker.component.html [hunk 1]
```diff
@@ -231,7 +231,7 @@
               'sky-btn sky-btn-link': 'modern'
             }"
             [style.backgroundColor]="color"
-            (click)="setColorFromString(color)"
+            (click)="onPresetClick(color)"
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/colorpicker/colorpicker.component.html:234]: Colorpicker was in dire need of some updates. To make sense of what was going on, I renamed some methods to match our current patterns. ☝️

File: src/app/public/modules/colorpicker/colorpicker.component.ts [hunk 2]
```diff
@@ -208,7 +209,7 @@ export class SkyColorpickerComponent implements OnInit, OnDestroy {
     private coreAdapter: SkyCoreAdapterService,
     private overlayService: SkyOverlayService,
     private service: SkyColorpickerService,
-    private themeSvc: SkyThemeService
+    @Optional() private themeSvc?: SkyThemeService
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/colorpicker/colorpicker.component.ts:212]: ☝️ Not part of the scope of this PR, but `themeSvc` should always be optional.

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-colorpicker?_sp.spa=pr-cancel-fix](https://developer.blackbaud.com/skyux-colorpicker?_sp.spa=pr-cancel-fix)

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-colorpicker?_sp.spa=pr-cancel-fix](https://developer.blackbaud.com/skyux-colorpicker?_sp.spa=pr-cancel-fix)

**Comment by Blackbaud-AdamFunderburk** [comment]
Can't reproduce the issue in the demo anymore. Nice work.

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-colorpicker?_sp.spa=pr-cancel-fix](https://developer.blackbaud.com/skyux-colorpicker?_sp.spa=pr-cancel-fix)

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Review by johnhwhite — APPROVED** [review]

**Pull request merged**